cache-config/tm-health-client/tmutil/tmutil.go

```diff
@@ -0,0 +1,674 @@
+// down in the trafficserver subsystem based upon that hosts current status and
+// the status that trafficmonitor health protocol has determined for a parent
+func (c *ParentInfo) PollAndUpdateCacheStatus() {
+	pollingInterval := config.GetTMPollingInterval()
+	log.Infoln("polling started")
+
+	for {
+		if err := c.UpdateParentInfo(); err != nil {
+			log.Errorf("could not load new ATS parent info: %s\n", err.Error())
+		} else {
+			log.Debugf("updated parent info, total number of parents: %d\n", len(c.Parents))
+		}
+
+		// read traffic manager cache statuses
+		caches, err := c.GetCacheStatuses()
+		if err != nil {
+			log.Errorln(err.Error())
+		}
+
+		for k, v := range caches {
+			hostName := string(k)
+			cs, ok := c.Parents[hostName]
+			if ok {
+				tmAvailable := *v.CombinedAvailable
+				if cs.available() != tmAvailable {
+					if !c.Cfg.EnableActiveMarkdowns {
+						if !tmAvailable {
+							log.Infof("TM reports that %s is not available and should be marked DOWN but, mark downs are disabled by configuration", hostName)
+						} else {
+							log.Infof("TM reports that %s is available and should be marked UP but, mark up is disabled by configuration", hostName)
+						}
+					} else {
+						if err = c.markParent(cs.Fqdn, *v.Status, tmAvailable); err != nil {
+							log.Errorln(err.Error())
+						}
+					}
+				}
+			}
+		}
+
+		time.Sleep(pollingInterval)
+	}
+}
+
+// Used by the polling function to update the parents list from
+// changes to parent.config and strategies.yaml.  The parents
+// availability is also updated to reflect the current state from
+// the trafficserver HostStatus subsystem.
+func (c *ParentInfo) UpdateParentInfo() error {
+	ptime, err := getFileModificationTime(c.ParentDotConfig.Filename)
+	if err != nil {
+		return errors.New("error reading parent.config: " + err.Error())
+	}
+	stime, err := getFileModificationTime(c.StrategiesDotYaml.Filename)
+	if err != nil {
+		return errors.New("error reading strategies.yaml: " + err.Error())
+	}
+	if c.ParentDotConfig.LastModifyTime < ptime {
+		// read parent.config
+		if err := c.readParentConfig(c.Parents); err != nil {
+			return errors.New("updating parent.config file: " + err.Error())
+		} else {
+			log.Infof("updated parents from new parent.config, total parents: %d\n", len(c.Parents))
+		}
+	}
+
+	if c.StrategiesDotYaml.LastModifyTime < stime {
+		// read strategies.yaml
+		if err := c.readStrategies(c.Parents); err != nil {
+			return errors.New("updating parent strategies.yaml file: " + err.Error())
+		} else {
+			log.Infof("updated parents from new strategies.yaml, total parents: %d\n", len(c.Parents))
+		}
+	}
+
+	// collect the trafficserver current host status
+	if err := c.readHostStatus(c.Parents); err != nil {
+		return errors.New("reading latest trafficserver host status: " + err.Error())
+	}
+
+	return nil
+}
+
+// choose an availabe trafficmonitor, returns an error if
+// there are none
```

Comment:
Nitpick: needs a period, GoDoc comments should be complete sentences/paragraphs.

Reply:
done